3-async-await/solution.js

```diff
@@ -19,17 +19,83 @@ Example:
 8. add any needed adjustment to solution() function
 9. as extra challenge: add Promise.race() and Promise.any(), and try to get the idea of what happens
 */
+const prices = require("./prices.js");
+const products = require("./products.js");
 
-function solution() {
-    // YOUR SOLUTION GOES HERE
+async function solution() {
+    // is this a bad practice? declaring but no initalization of variables that will eventually have a value in all logic branches?
+    let allResult, settledResult, raceResult, anyResult;
 
     // You generate your id value here
+    const rng = parseInt(Date.now().toString().slice(-2));
+
+    // Yes, here we generate new promises for each method, 
+    // we could remove the awaits and wrap the promises in another promise
+    // but that is too meta for this exercise for now. 
 
     // You use Promise.all() here
+    try {
+        allResult = await Promise.all([prices(rng), products(rng)]);
```

Comment:
in this case it is better to destructure the array since you already know which is index 0 and index 1
```suggestion
       const [price, product] = await Promise.all([prices(rng), products(rng)]);
```